lib/IDE/CodeCompletion.cpp

```diff
@@ -5167,9 +5170,26 @@ class CodeCompletionTypeContextAnalyzer {
   SmallVector<FunctionTypeAndDecl, 2> PossibleCallees;
 
   void recordPossibleType(Type ty) {
+    if (ty) {
+      ty = ty->getRValueType();
+
+      // Convert 'GenericTypeParamType' to 'ArchetypeType'.
+      if (ty->hasTypeParameter()) {
```

Comment:
I think you can replace this whole block with `DC->mapTypeIntoContext(ty)`. Please don't use GenericTypeParamType::getDecl().